yktrace/src/debug.rs

```diff
@@ -48,20 +48,38 @@ pub fn def_path(def_id: &DefId) -> Option<&str> {
 }
 
 /// Prints a SIR trace to stdout for debugging purposes.
-pub fn print_sir_trace(trace: &dyn SirTrace) {
+pub fn print_sir_trace(trace: &dyn SirTrace, trimmed: bool) {
+    let locs: Vec<&SirLoc> = match trimmed {
+        false => (0..(trace.raw_len())).map(|i| trace.raw_loc(i)).collect(),
+        true => trace.into_iter().collect()
+    };
+
     println!("---[ BEGIN SIR TRACE DUMP ]---");
-    for loc in trace {
+    for loc in locs {
         let def_id = DefId::from_sir_loc(&loc);
         let def_path_s = match def_path(&def_id) {
             Some(s) => s,
             None => "<unknown>"
         };
+
+        let body = SIR.bodies.get(&def_id);
+        let mut flags = Vec::new();
```

Comment:
This intermediate `Vec` feels like a real song and dance for nothing. Why not move the `if` after the `println` and just `println("HEAD")` (or `TAIL`) as appropriate?

Reply:
Because there may be more than one flag per trace location.

Reply:
OK... but then still just do it as a series of `print`s after the one that's already there maybe?

Reply:
Yeah, I suppose we could do. I didn't put a great deal of effort into optimising this, as it's debug code that won't be called in production.

Reply:
Understood. At the moment the intermediate list is just a bit of a distraction, so I think simplifying is a good idea.